File: api/src/main/java/com/example/nb_weather_ol/controller/WeatherDataController.java
package com.example.nb_weather_ol.controller;

import com.example.nb_weather_ol.model.*;
import com.example.nb_weather_ol.service.WeatherDataService;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/wd")
public class WeatherDataController {

    private final WeatherDataService weatherDataService;

    @Autowired
    public WeatherDataController(WeatherDataService weatherDataService) {
        this.weatherDataService = weatherDataService;
    }

    @PostMapping
    //
    public ResponseEntity<List<WeatherData>> createWeatherData(@RequestBody WeatherDataRequest request) {
        try {
            List<WeatherData> createdData = weatherDataService.createWeatherData(request.getQuery(),
                    request.getQueryId());
            return new ResponseEntity<>(createdData, HttpStatus.CREATED);
        } catch (RuntimeException e) {
            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
        }
    }

    @GetMapping("/{id}")
    public ResponseEntity<List<WeatherData>> getWeatherDataById(@PathVariable Long id) {
        try {
            List<WeatherData> weatherDataList = weatherDataService.getWeatherDataById(id);
            if (weatherDataList.isEmpty()) {
                return new ResponseEntity<>(HttpStatus.NOT_FOUND);
            }
            return new ResponseEntity<>(weatherDataList, HttpStatus.OK);
        } catch (RuntimeException e) {
            return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
        }
    }

    @GetMapping
    public ResponseEntity<List<WeatherData>> getAllWeatherData() {
        try {
            List<WeatherData> allData = weatherDataService.getAllWeatherData();
            return new ResponseEntity<>(allData, HttpStatus.OK);
        } catch (RuntimeException e) {
            return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
        }
    }
}

File: backend/src/main/kotlin/com/nbweatherproject/backend/controller/WeatherDataController.kt
package com.nbweatherproject.backend

import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/wd")
class WeatherDataController(private val weatherDataService: WeatherDataService) {

    @PostMapping
    fun createWeatherData(@RequestBody request: WeatherDataRequest): ResponseEntity<List<WeatherData>> {
        return try {
            val createdData = weatherDataService.createWeatherData(request.query, request.queryId)
            ResponseEntity(createdData, HttpStatus.CREATED)
        } catch (e: RuntimeException) {
            ResponseEntity(HttpStatus.BAD_REQUEST)
        }
    }

    @GetMapping("/i/{id}")
    fun getWeatherDataById(@PathVariable id: Long): ResponseEntity<List<WeatherData>> {
        return try {
            val weatherDataList = weatherDataService.getWeatherDataById(id)
            if (weatherDataList.isEmpty()) {
                ResponseEntity(HttpStatus.NOT_FOUND)
            } else {
                ResponseEntity(weatherDataList, HttpStatus.OK)
            }
        } catch (e: RuntimeException) {
            ResponseEntity(HttpStatus.INTERNAL_SERVER_ERROR)
        }
    }

    @GetMapping
    fun getAllWeatherData(): ResponseEntity<List<WeatherData>> {
        return try {
            val allData = weatherDataService.getAllWeatherData()
            ResponseEntity(allData, HttpStatus.OK)
        } catch (e: RuntimeException) {
            ResponseEntity(HttpStatus.INTERNAL_SERVER_ERROR)
        }
    }
}